### PR TITLE
Fixed Flaky Test in cypress develop branch

### PR DESCRIPTION
**Pull request by nihal467**
## Proposed Changes

- Fixes #issue_number
- Change 1
- Change 2
- More?

@ohcnetwork/care-fe-code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

## Summary by CodeRabbit

- **New Features**
	- Enhanced the `clickAndSelectOption` command with an optional parameter to skip verification.
	- Introduced utility functions for generating patient names and random addresses.
	- Added a constant for modified patient categories in log updates.
	- Added new methods to verify round type and patient category in health metrics entry.

- **Bug Fixes**
	- Improved reliability of the `clickAndSelectOption` command by confirming expected outcomes post-interaction.
	- Updated methods for entering various health metrics to enhance input verification robustness.
	- Refined patient registration tests to include checks for error messages during transfers.
	- Adjusted patient consultation process to target the correct element for interaction.
	- Modified the handling of systolic and diastolic values during log updates for improved input processing.
	- Increased wait times in advance filter methods to ensure smoother operation.

- **Chores**
	- Removed outdated methods related to patient URL handling to streamline navigation management.

### CODE REVIEW

File: cypress/support/commands.ts [hunk 1]
```diff
@@ -133,12 +133,12 @@ Cypress.Commands.add("clickCancelButton", (buttonText = "Cancel") => {
...
-  (element: string, referance: string) => {
+  (element: string, reference: string) => {
     cy.get(element)
       .click()
-      .type(referance)
+      .type(reference)
       .then(() => {
-        cy.get("[role='option']").contains(referance).click();
+        cy.get("[role='option']").contains(reference).click();
       });
```
[the remark this answers is not in the document]
Comment (nihal467) [reply cypress/support/commands.ts:142]: sometimes we dont type completely or something more will be there in the entire word, so lets dont add it now

File: cypress/e2e/patient_spec/PatientRegistration.cy.ts [hunk 2]
```diff
@@ -149,13 +152,12 @@ describe("Patient Creation with consultation", () => {
...
-  it("Edit the patient details with no consultation and verify", () => {
-    patientPage.interceptFacilities();
-    patientPage.visitUpdatePatientUrl();
-    patientPage.verifyStatusCode();
-    patientPage.patientformvisibility();
+  it("Edit the patient details and verify its reflection", () => {
+    const patientName = "Dummy Patient Two";
+    patientPage.visitPatient(patientName);
+    patientConsultationPage.clickPatientDetails();
+    patientPage.clickPatientUpdateDetails();
```
[the remark this answers is not in the document]
Comment (nihal467) [reply cypress/e2e/patient_spec/PatientRegistration.cy.ts:143]: i want the test to interact with a particular dummy data patient, so its not possible

[the remark this answers is not in the document]
Comment (nihal467) [reply cypress/e2e/patient_spec/PatientRegistration.cy.ts:143]: i want it to interact with specific dumy data

File: cypress/pageobject/Patient/PatientConsultation.ts [hunk 3]
```diff
@@ -65,7 +65,7 @@ export class PatientConsultationPage {
   }
 
   clickPatientDetails() {
-    cy.verifyAndClickElement("#consultationpage-header", "Patient Details");
+    cy.verifyAndClickElement("#patient-details", "Patient Details");
```
[the remark this answers is not in the document]
Comment (nihal467) [reply cypress/pageobject/Patient/PatientConsultation.ts:68]: will look into it later

**Review by khavinshankar — APPROVED** [review]

**Pull request merged**